src/detect-env.ts

```diff
@@ -29,18 +29,16 @@ function runEnvChecks(...strategies: EnvCheckStrategy[]) {
 /**
  * Runs checks that determine if user is using preact.
  * So far they are not ideal, as there is no consistent way to detect preact.
- * Right now the main determinant is if synthetic event was not detected.
  * */
 function isPreact(context: DetectEnvContext) {
-  return !context.syntheticEventDetected
+  return context.classRenderReceivesAnyArguments
 }
 
 /**
  * Checks if user is using react.
- * So far we are doing that by checking if synthetic event was detected.
  * */
 function isReact(context: DetectEnvContext) {
-  return context.syntheticEventDetected
+  return !isPreact(context)
```

Comment:
Not sure if we want to be more explicit here and have something like:
```suggestion
  return !context.classRenderReceivesAnyArguments
```

Reply:
Good, done.